### PR TITLE
Automated deployment for extension on tag

### PR DESCRIPTION
**Pull request by inverse**
This change should enable us to automatically deploy via travis to the chrome store when we tag a release.

### Changes
Will require setting up some environment variables on Travis to contain the secrets

#### Chrome
`CHROME_CLIENT_ID`
`CHROME_CLIENT_SECRET`
`CHROME_REFRESH_TOKEN`

#### Github
 `GITHUB_TOKEN`

#### Amo
 `AMO_ISSUER`
 `AMO_SECRET`

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+./node_modules/.bin/grunt publish:chrome
```
Comment (inverse) [inline deploy.sh]: @alex-volga can you please add the commands we do for a release. e.g. bump/dump etc

File: Gruntfile.js [hunk 2]
```diff
@@ -427,4 +427,46 @@ module.exports = function(grunt) {
 
 		return {};
 	}
+
+	function loadWebStoreConfig() {
+		if (isTravisCi) {
```
Comment (inverse) [inline Gruntfile.js]: This is so we can use the "old way" but if we are on travis do it travis style ;)

**Comment by stephanebruckert** [comment]
This is awesome! ping @alexesprit

**Comment by alexesprit** [comment]
The `release` task should not run publish tasks. In my vision with automatic deployment you want to run `release` task to make a release. The task will create a "bump" commit, a new tag contains a version number, and will push changes (IIRC, there's no push now, but it can be added). And Travis CI will do the rest work, i.e. deploy packages via `publish` tasks on tag create.

Everything else looks fine for me.

**Comment by inverse** [comment]
@alexesprit right now I'm calling `./node_modules/.bin/grunt publish:chrome` on the `deploy.sh` file. Is there anything else that needs to be included here?

**Comment by inverse** [comment]
@alexesprit How would that tag align with what's already been drafted on GH?

So are you saying the `release` task needs to be reworked to just `bump` and `tag`? I understand the version needs to be bumped before tagging.

**Comment by alexesprit** [comment]
> How would that tag align with what's already been drafted on GH?

It's only thing I have no idea how to automate. You should manually assign a new tag to a draft after pushing this tag. I used this way when I released new versions.

> So are you saying the release task needs to be reworked to just bump and tag? I understand the version needs to be bumped before tagging.

Yes, you're right. Otherwise, the package will be uploaded twice. Of course, you can bump and push things manually, and in this case the package will be uploaded by Travis CI only.

> We still need you to put the credentials into env vars on travis to automate the deployment.

I understand this; I've managed how to add env vars to Travis CI project. I'll generate new credentials and set up the project. I'll let you know when I do it, so we can release new version and see how the automatic things work.

**Comment by inverse** [comment]
My assumption was that it would work like this.

- Develop + push
- Bump + push
- Tag - on GH
- Build kicked off on travis

or am I mistaken?

**Comment by alexesprit** [comment]
1. Develop + push changes.
2. When the release is ready to publish, you run `grunt release`, which bumps version, add new tag and push changes to GitHub.
3. Travis CI detects new tag and deploys packages.

**Comment by alexesprit** [comment]
> (IIRC, there's no push now, but it can be added)

Just found out the default configuration is to push changes to upstream after bump and commit. I didn't change this, so on `release` task changes will be pushed to upstream.

**Comment by inverse** [comment]
I added a new task called `release-local` which will provide the same experience as release but with the publish steps included. It will however cause another build to be triggered from travis with the tag.

There is one final part I am not sure about and that's the publish addon part. For me this feels like it should be part of the deployment. e.g. hooked up in `deploy.sh`. What do you think?

**Comment by inverse** [comment]
I've merged this for now - we can discuss how to handle the addon part later.

**Comment by inverse** [comment]
My bad - I understand it now. Changed back with: https://github.com/web-scrobbler/web-scrobbler/pull/1776

**Pull request merged**